modules/README.md

```diff
@@ -237,6 +237,24 @@ function encodeData(bytes32 domain, bytes32 hashStruct, bytes calldata signature
 }
 ```
 
+In the default signature validation flow, the wallet is expected to generate a hash of the message i.e., `messageHash` for signing and submit it for validation as follows:
+
+```
+    ACCOUNT_MSG_TYPEHASH = keccak256("SafeMessage(bytes message)");
+    DOMAIN_SEPARATOR_TYPEHASH = keccak256(EIP712Domain(uint256 chainId,address verifyingContract));
+
+    domainSeparator = keccak256(DOMAIN_SEPARATOR_TYPEHASH || chainId || accountAddress);
+    dataHash = keccak256("encoded-bytes-data-to-be-signed");
+
+    hashStruct = keccak256(DOMAIN_SEPARATOR_TYPEHASH || dataHash);
+    messageHash = keccak256(0x19 || 0x01 || domainSeparator || dataHash);
+
+    signatures = generateSignaturesUsingWallet(messageHash);
+    result = await account.isValidSignature.staticCall(dataHash, signatures);
```

Comment:
Since this is not a proper javascript code I'd remove the `await` and `.staticCall`